apps/examples/testcase/le_tc/filesystem/itc_fs.c

```diff
@@ -0,0 +1,1349 @@
+
+	rewinddir(dirp);
+
+	st_dirent = readdir(dirp);
+	TC_ASSERT_EQ_CLEANUP("readdir", st_dirent, NULL, closedir(dirp); rmdir(VFS_FOLDER_PATH));
+
+	ret = closedir(dirp);
+	TC_ASSERT_EQ_CLEANUP("closedir", ret, OK, rmdir(VFS_FOLDER_PATH));
+
+	dirp = opendir(VFS_FOLDER_PATH);
+	TC_ASSERT_NEQ_CLEANUP("opendir", dirp, NULL, rmdir(VFS_FOLDER_PATH));
+
+	ret = mkdir(folder_name_0, 0777);
+	TC_ASSERT_EQ_CLEANUP("mkdir", ret, OK, closedir(dirp); rmdir(VFS_FOLDER_PATH));
+
+	st_dirent = readdir(dirp);
+	TC_ASSERT_NEQ_CLEANUP("readdir", st_dirent, NULL, closedir(dirp); remove_dirs(count));
+
+	ret = mkdir(folder_name_1, 0777);
+	TC_ASSERT_EQ_CLEANUP("mkdir", ret, OK, closedir(dirp); remove_dirs(count));
+
+	ret = mkdir(folder_name_2, 0777);
+	TC_ASSERT_EQ_CLEANUP("mkdir", ret, OK, closedir(dirp); remove_dirs(count));
+
+	rewinddir(dirp);
+
+	ret = rmdir(folder_name_0);
+	TC_ASSERT_EQ_CLEANUP("rmdir", ret, OK, closedir(dirp); remove_dirs(count));
+
+	while (readdir(dirp) != NULL) {
+		dir_count++;
+	}
+
+	TC_ASSERT_EQ_CLEANUP("readdir", dir_count, 2, closedir(dirp); remove_dirs(count));
+
+	ret = closedir(dirp);
+	TC_ASSERT_EQ_CLEANUP("closedir", ret, OK, remove_dirs(count));
+
+	ret = rmdir(folder_name_1);
+	TC_ASSERT_EQ_CLEANUP("rmdir", ret, OK, remove_dirs(count));
+
+	ret = rmdir(folder_name_2);
+	TC_ASSERT_EQ_CLEANUP("rmdir", ret, OK, remove_dirs(count));
+
+	ret = rmdir(VFS_FOLDER_PATH);
+	TC_ASSERT_EQ("rmdir", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_fs_vfs_seekdir_telldir_p
+* @brief            Move position of contents in directory, Get position of contents in directory.
+* @scenario         move to different locations, and get the location.
+* @apicovered       seekdir, telldir
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_fs_vfs_seekdir_telldir_p(void)
+{
+	DIR *dirp;
+	struct dirent *st_dirent;
+	off_t offset;
+	int count;
+	char buffer[BUFFER_LEN];
+
+	g_flag = OK;
+	add_dirs(VFS_LOOP_COUNT);
+	TC_ASSERT_EQ("mkdir", g_flag, OK);
+
+	dirp = opendir(VFS_FOLDER_PATH);
+	TC_ASSERT_NEQ_CLEANUP("opendir", dirp, NULL, remove_dirs(VFS_LOOP_COUNT));
+
+	TC_ASSERT_EQ_CLEANUP("telldir", telldir(dirp), (off_t)0, closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+
+	count = 0;
+	while (1) {
+		offset = count;
+		seekdir(dirp, offset);
+		TC_ASSERT_NEQ_CLEANUP("seekdir", dirp, NULL, closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+
+		if (count < 0) {
+			TC_ASSERT_EQ_CLEANUP("telldir", telldir(dirp), (off_t)0, closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+			st_dirent = readdir(dirp);
+			TC_ASSERT_NEQ_CLEANUP("readdir", st_dirent, NULL, closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+			TC_ASSERT_EQ_CLEANUP("readdir", st_dirent->d_name[0], '0', closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+			TC_ASSERT_EQ_CLEANUP("telldir", telldir(dirp), (off_t)1, closedir(dirp); remove_dirs(VFS_LOOP_COUNT));
+		} else if (count < 5) {
```

Comment:
To understand easily, (count >= 0 && count < 5) will be good.

Reply:
@Taejun-Kwon 
Fixed Review comments. Please verify.